### PR TITLE
Prevent closure allocations in OnHeartbeat

### PR DESCRIPTION
**Pull request by pakrym**
Should prevent memory growing in https://github.com/aspnet/KestrelHttpServer/issues/1260

@halter73 @cesarbs

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs [hunk 1]
```diff
@@ -23,6 +23,11 @@ public class KestrelThread
 
         private static readonly Action<object, object> _postCallbackAdapter = (callback, state) => ((Action<object>)callback).Invoke(state);
         private static readonly Action<object, object> _postAsyncCallbackAdapter = (callback, state) => ((Action<object>)callback).Invoke(state);
+        private static readonly Libuv.uv_walk_cb _heartbeatWalkCallback = (ptr, arg) =>
```
Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs]: nit: call it `_heartbeatWalkCallbackAdapter` to be consistent with the other ones.

**Comment by halter73** [comment]
`nit: call it _heartbeatWalkCallbackAdapter to be consistent with the other ones.` I think it show go back to `_heartbeatWalkCallback`. Unlike the other `Adapter`s, it's not calling a callback passed in as an argument.

**Comment by cesarbs** [comment]
:shipit:

**Comment by cesarbs** [comment]
@halter73 You're right, I overlooked that. @pakrym Sorry, revert the last commit :grin:

**Comment by cesarbs** [comment]
Squash and :rocket:

**Pull request merged**